src/main/java/frc/robot/HardwareMap.java

```diff
@@ -13,6 +17,25 @@ public final class HardwareMap {
 	public static final int CAN_ID_SPARK_SHOOTER = 5;
 
 	/* ===== Hardware Availability ===== */
+
+	/**
+	 * Check if the hardware for the given systsem is available to the RoboRIO.
+	 * @param type the requested system in check for hardware presence
+	 * @return true if system hardware is present
+	 */
+	public static boolean isSystemHardwarePresent(Class<?> type) {
```

Comment:
I'd prefer to not use class introspection when we know exactly how many systems we have. Introspection is brittle and difficult to understand. We may change the presence checks to read a digital input at some point.